### PR TITLE
Update action's output to include 3 fields.

### PR DESCRIPTION
**Pull request by gusingh-r7**
## Proposed Changes

Describe the proposed changes:

  - Update action's output to include 3 fields i.e. STDERR, STDOUT, STDERR_STDOUT

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `./run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `./run -T tests/example.json -d -j`
    * All actions/triggers shortcut `./run -T all -d -j` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `./run -R tests/example.json -d -j`
    * All actions/triggers shortcut `./run -R all -d -j` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `./run -A -R all -T all`. For single action validation: `./run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`./run ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by gusingh-r7** [comment]
## Assessment
### Run

<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via key\n",
    "meta": {},
    "output": {
      "results": {
        "stderr": "",
        "stdout": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n",
        "stdout_stderr": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 run < tests/run_key.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via password\n",
    "meta": {},
    "output": {
      "results": {
        "stderr": "",
        "stdout": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n",
        "stdout_stderr": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 run < tests/run_key_file.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via password\n",
    "meta": {},
    "output": {
      "results": {
        "stderr": "",
        "stdout": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n\nUSER       PID %!C(MISSING)PU %!M(MISSING)EM    VSZ   RSS TTY      STAT START   TIME COMMAND\n",
        "stdout_stderr": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n\nUSER       PID %!C(MISSING)PU %!M(MISSING)EM    VSZ   RSS TTY      STAT START   TIME COMMAND\n"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 run < tests/run_multiple_commands.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via password\n",
    "meta": {},
    "output": {
      "results": {
        "stderr": "",
        "stdout": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n",
        "stdout_stderr": "total 57068\n\ndrwxrwxr-x. 2 rapid7 rapid7        6 Nov  4 20:15 test\n\n-rw-rw-r--. 1 rapid7 rapid7       13 Nov  4 20:15 test.txt\n\n-rw-r--r--. 1 rapid7 rapid7 58433536 Mar 26  2019 VBoxGuestAdditions.iso\n"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 run < tests/run_pass.json
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via key\n",
    "meta": {},
    "output": {
      "connection": "successful"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 test < tests/run_key.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via password\n",
    "meta": {},
    "output": {
      "connection": "successful"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 test < tests/run_key_file.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via password\n",
    "meta": {},
    "output": {
      "connection": "successful"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 test < tests/run_multiple_commands.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "connecting\nrapid7/SSH:2.0.0. Step name: run\nconnecting via password\n",
    "meta": {},
    "output": {
      "connection": "successful"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/ssh:2.0.0 test < tests/run_pass.json
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.

#### Workflow Builder


#### Job


#### Artifact

File: ssh/plugin.spec.yaml [hunk 1]
```diff
@@ -16,6 +16,23 @@ tags:
...
+      description: Stdout
+      type: string
+      required: true
+    stderr:
+      title: STDERR
+      description: Stderr
+      type: string
+      required: true
+    stdout_stderr:
+      title: Stdout Stderr
```
Comment (mhofert-r7) [inline ssh/plugin.spec.yaml]: Titles need to be standardized. ie `stdout` the title is STDOUT, but for `stdout_stderr` the title is Stdout Stderr. The titles need to be all caps or only the first letter as caps. I'm not sure which is correct

Comment (lmilby-r7) [reply ssh/plugin.spec.yaml]: Maybe `STDOUT & STDERR`

Comment (jschipp-r7) [reply ssh/plugin.spec.yaml]: Could call this `All Output`

File: ssh/plugin.spec.yaml [hunk 2]
```diff
@@ -16,6 +16,23 @@ tags:
...
+      title: STDOUT
+      description: Stdout
+      type: string
+      required: true
+    stderr:
+      title: STDERR
+      description: Stderr
+      type: string
+      required: true
+    stdout_stderr:
```
Comment (jschipp-r7) [inline ssh/plugin.spec.yaml]: I think we should call this All Output, it sound a lot better IMO.

File: ssh/help.md [hunk 3]
```diff
@@ -95,3 +99,13 @@ Examples:
...
+
+## Custom Output Types
+
+### results
+
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|stderr|string|True|Stderr|
+|stdout|string|True|Stdout|
+|stdout_stderr|string|True|All output|
```
Comment (jschipp-r7) [inline ssh/help.md]: Update variable name in table

Comment (jschipp-r7) [reply ssh/help.md]:
```suggestion
|all_output|string|True|All output|
```

File: ssh/help.md [hunk 4]
```diff
@@ -88,9 +92,20 @@ Examples:
 * 1.0.0 - Support web server mode | Update to new credential types | Rename "Run remote command" action to "Run Remote Command"
 * 1.0.1 - Fix issue where run action was excluded from plugin on build
 * 1.0.2 - Fixed issue where Run was excluded
+* 2.0.0 - Update action's output to include 3 fields i.e. STDERR, STDOUT, STDERR_STDOUT
```
Comment (jschipp-r7) [inline ssh/help.md]: Update changelog too to 

```suggestion
* 2.0.0 - Update action's output to include 3 output fields i.e. `stderr`, `stdout`, and `all_output`
```

File: ssh/help.md (hunk 3 above)
Comment (jschipp-r7) [reply ssh/help.md]: @gusingh-r7 Apply this suggestion and we can merge

**Pull request merged**